test/integration_tests/python/test_containers.py

```diff
@@ -126,3 +126,140 @@ def test_project_template(with_hierarchy, data_builder, as_user):
+    r = as_public.get('/projects', params={'info': 'true'})
+    assert r.ok
+
+    # get all projects w/ counts=true
+    r = as_public.get('/projects', params={'counts': 'true'})
+    assert r.ok
+    assert all('session_count' in proj for proj in r.json())
+
+    # get all sessions for project w/ measurements=true and stats=true
+    r = as_public.get('/projects/' + data.project + '/sessions', params={
+        'measurements': 'true',
+        'stats': 'true'
+    })
+    assert r.ok
+
+
+def test_get_all_for_user(as_user, as_public):
+    r = as_user.get('/users/self')
+    user_id = r.json()['_id']
+
+    # try to get containers for user w/o logging in
+    r = as_public.get('/users/' + user_id + '/sessions')
+    assert r.status_code == 403
+
+    # get containers for user
+    r = as_user.get('/users/' + user_id + '/sessions')
+    assert r.ok
+
+
+def test_get_container(with_hierarchy, as_user, as_public):
+    data = with_hierarchy
+
+    # NOTE cannot reach APIStorageException - wanted to cover 400 error w/ invalid oid
+    # but then realized that api.py's cid regex makes this an invalid route resulting in 404
+
+    # try to get container w/ invalid object id
+    # r = as_user.get('/projects/test')
+    # assert r.status_code == 400
+
+    # try to get container w/ nonexistent object id
+    r = as_public.get('/projects/000000000000000000000000')
+    assert r.status_code == 404
+
+    # get container
+    r = as_public.get('/projects/' + data.project)
+    assert r.ok
+
+    # get container w/ ?paths=true
+    r = as_user.post('/projects/' + data.project + '/files', files={
+        'file': ('test.csv', 'header\nrow1\n'),
+        'metadata': ('', json.dumps({'name': 'test.csv', 'type': 'csv'}))
+    })
+    assert r.ok
+
+    r = as_public.get('/projects/' + data.project, params={'paths': 'true'})
+    assert r.ok
+    assert all('path' in f for f in r.json()['files'])
+
+    # get container w/ ?join=origin
+    r = as_public.get('/projects/' + data.project, params={'join': 'origin'})
+    assert r.ok
+    assert 'join-origin' in r.json()
+
+
+def test_get_session_jobs(with_hierarchy, with_gear, as_user, db):
+    data = with_hierarchy
+    gear = with_gear
+
+    # get session jobs w/ analysis and job
+    r = as_user.post('/sessions/' + data.session + '/analyses', params={'job': 'true'}, json={
+        'analysis': {'label': 'test analysis'},
+        'job': {
+            'gear_id': gear,
+            'inputs': {
+                'dicom': {
+                    'type': 'acquisition',
+                    'id': data.acquisition,
+                    'name': 'test.dcm'
+                }
+            }
+        }
+    })
+    assert r.ok
+
+    r = as_user.get('/sessions/' + data.session + '/jobs', params={'join': 'containers'})
+    assert r.ok
+
+    # TODO figure out better/generalized way to handle state cleanup
+    # make sure other tests start clean w/o existing jobs (eg for /jobs/next to work as expected)
```

Comment:
I wouldn't be opposed to adding the removal of all jobs relating to this gear to the `with_gear` fixture's `teardown_db` method.